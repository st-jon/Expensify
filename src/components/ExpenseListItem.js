import React from 'react'
import { Link } from 'react-router-dom'
import moment from 'moment'

export const ExpenseListItem = ({ id, description, amount, createdAt}) => ( 
    <div>
        <Link to={`/edit/${id}`}>
            <h3>{description}</h3>
        </Link>
            <p>
            {new Intl.NumberFormat('de-DE', { style: 'currency', currency: 'EUR' }).format(amount / 100)}
            // 
            {moment(createdAt).locale('fr').format('LL')}
            </p>
    </div>
)

export default ExpenseListItem