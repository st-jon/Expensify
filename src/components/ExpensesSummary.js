import React from 'react'
import { connect } from 'react-redux'
import selectExpenses from '../selectors/expenses'
import selectExpensesTotal from '../selectors/expenses-total'

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
    const expenseWord = expenseCount === 1 ? 'expense' : 'expenses'
    const formattedExpensesTotal = new Intl.NumberFormat('de-DE', { style: 'currency', currency: 'EUR' }).format(expensesTotal / 100)
    return (
        <div>
            <p>Viewing {expenseCount} {expenseWord} for a total amount of {formattedExpensesTotal}.</p>
        </div>
    )
}

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters)
    return {
        expenseCount: visibleExpenses.length,
        expensesTotal: selectExpensesTotal(visibleExpenses)
    }
}

export default connect(mapStateToProps)(ExpensesSummary)