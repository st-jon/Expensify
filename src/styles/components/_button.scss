.button {
    background: $dark-blue;
    border: none;
    color: fade-out(white, .15);
    display: inline-block;
    font-size: $font-size-large;
    font-weight: 300;
    line-height: 1;
    padding: $s-size;
    text-decoration: none;
}

.button--link {
    background: none;
}

.button--secondary {
    background: #888888
}