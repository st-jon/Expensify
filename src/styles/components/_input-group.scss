.input-group {
    display: flex;
    flex-direction: column;
    margin-bottom: $m-size;
    padding: $s-size 0;

    @media (min-width: $desktop-breakpoint) {
        flex-direction: row;
        margin-bottom: $l-size;
    }
}

.input-group__item {
    margin-bottom: $s-size;
    margin-right: $s-size;

    @media (min-width: $desktop-breakpoint) {
        margin-right: 0 $s-size 0 0;
    }
}