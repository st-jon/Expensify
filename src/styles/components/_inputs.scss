.text-input {
    border: 1px solid rgb(219, 219, 219);
    height: 50px;
    padding: $s-size;
    font-size: $font-size-large;
    font-weight: 300;
}

.select {
    @extend .text-input;   
}

.textarea {
    @extend .text-input;
    height: 10rem;
}