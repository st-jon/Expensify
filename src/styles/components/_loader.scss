.loader {
    align-items: center;
    display: flex;
    justify-content: center;
    height: 100vh;
    width: 100vw;

}

.loader__image {
    height: 5rem;
    width: 5rem;
}