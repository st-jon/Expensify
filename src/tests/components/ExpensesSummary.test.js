import React from 'react'
import { shallow } from 'enzyme'
import { ExpensesSummary } from '../../components/ExpensesSummary'

test('Should render ExpensesSummary with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={235} />)
    expect(wrapper).toMatchSnapshot()
})

test('Should render ExpensesSummary with multiple expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={23} expensesTotal={23785865} />)
    expect(wrapper).toMatchSnapshot()
})