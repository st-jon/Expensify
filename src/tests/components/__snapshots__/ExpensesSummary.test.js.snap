// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpensesSummary with 1 expense 1`] = `
<div>
  <p>
    Viewing 
    1
     
    expense
     for a total amount of 
    €2.35
    .
  </p>
</div>
`;

exports[`Should render ExpensesSummary with multiple expense 1`] = `
<div>
  <p>
    Viewing 
    23
     
    expenses
     for a total amount of 
    €237,858.65
    .
  </p>
</div>
`;
